examples/defaulter-gen/generators/defaulter.go

```diff
@@ -803,15 +804,40 @@ func (g *genDefaulter) GenerateType(c *generator.Context, t *types.Type, w io.Wr
 
 			// Rewrite the fully qualified name using the local package name
 			// from the imports
-			localPackage := g.imports.LocalNameOf(parsedName.Package)
-			if len(localPackage) > 0 {
-				current.defaultValue.SymbolReference = localPackage + "." + parsedName.Name
+			if len(parsedName.Package) > 0 && g.isOtherPackage(parsedName.Package) {
```

Comment:
Fixes a bug where fully specifying a symbol in local package would cause package name to be used.